### PR TITLE
Populate SIGMA_SPECTRUM and redo SIGMA and WEIGHT

### PR DESCRIPTION
**Pull request by bennahugo**
Fixes SIGMA_SPECTRUM, WEIGHT and SIGMA column issues as discussed.

### CODE REVIEW

**Comment by bennahugo** [comment]
@ludwigschwardt can you review?

File: katdal/ms_extra.py [hunk 1]
```diff
@@ -372,6 +381,13 @@ def populate_main_dict(uvw_coordinates, vis_data, flag_data, weight_data, timest
     main_dict['FLAG_ROW'] = np.zeros(num_vis_samples, dtype=np.uint8)
     # The visibility weights
     main_dict['WEIGHT_SPECTRUM'] = weight_data
+    # Inverse visibility weights
+    # note this column is used when computing calibration weights
+    # in CASA - WEIGHT_SPECTRUM may be modified based on the
+    # values in this column. See
+    # https://casa.nrao.edu/casadocs/casa-5-1.2/reference-material/data-weights
```
Comment (ludwigschwardt) [inline katdal/ms_extra.py:388]: There is also this link which seems to be the latest version at all times (yours is associated with v5.1.2):

https://casadocs.readthedocs.io/en/stable/notebooks/data_weights.html

File: katdal/ms_extra.py [hunk 2]
```diff
@@ -372,6 +381,13 @@ def populate_main_dict(uvw_coordinates, vis_data, flag_data, weight_data, timest
     main_dict['FLAG_ROW'] = np.zeros(num_vis_samples, dtype=np.uint8)
     # The visibility weights
     main_dict['WEIGHT_SPECTRUM'] = weight_data
+    # Inverse visibility weights
```
Comment (ludwigschwardt) [inline katdal/ms_extra.py:384]: "Inverse square root visibility weights", just to be safe :-D

Or "Estimated RMS noise per frequency channel", like the SIGMA description.

File: katdal/ms_extra.py [hunk 3]
```diff
@@ -372,6 +381,13 @@ def populate_main_dict(uvw_coordinates, vis_data, flag_data, weight_data, timest
...
+    # note this column is used when computing calibration weights
+    # in CASA - WEIGHT_SPECTRUM may be modified based on the
+    # values in this column. See
+    # https://casa.nrao.edu/casadocs/casa-5-1.2/reference-material/data-weights
+    # for further details
+    main_dict['SIGMA_SPECTRUM'] = weight_data**-0.5
```
Comment (ludwigschwardt) [inline katdal/ms_extra.py:390]: Spaces around `**` operator.

File: katdal/ms_extra.py [hunk 4]
```diff
@@ -397,7 +414,7 @@ def populate_main_dict(uvw_coordinates, vis_data, flag_data, weight_data, timest
     main_dict['UVW'] = np.asarray(uvw_coordinates)
     # Weight for each polarisation spectrum (float, 1-dim). This is just
     # just filled with 1's, because the real weights are in WEIGHT_SPECTRUM.
```
Comment (ludwigschwardt) [inline katdal/ms_extra.py:416]: This comment is now outdated.

**Review by ludwigschwardt — CHANGES_REQUESTED** [review]
Great!

**Comment by bennahugo** [comment]
Alright, done as requested

**Review by ludwigschwardt — APPROVED** [review]

**Comment by ludwigschwardt** [comment]
This is to make up for all those multi-year overdue reviews 😂

**Pull request merged**